python-mini-projects/projects/Write_script_to_move_files_into_alphabetically_ordered_folder/main.py

```diff
@@ -0,0 +1,75 @@
+'''
+This script will sort and move the files in the directory
+(the alphabetical order).
+
+'apple.txt' --> 'A'
+'ryan.txt' --> 'R'
+'01010.txt' --> 'Misc'
+
+'''
+import os
+import shutil
+
+filenames = []
+
+
+def getfoldername(filename):
+    '''
+    'Test.txt' --> 't'
+    '010.txt' --> 'misc'
+    'zebra.txt' --> 'z'
+    'Alpha@@.txt' --> 'a'
+    '!@#.txt' --> 'misc'
+    '''
+    if filename[0].isalpha():
+        return filename[0].lower()
+    else:
+        return 'misc'
+
+
+def readdirectory():
+    '''
+    read the filename in the current directory and append them to a list
+    '''
+    global filenames
+    for files in os.listdir(os.getcwd()):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)